### PR TITLE
chore(sdk): fix test missing tempdir cleanup

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
This PR fixes a test that left a gs:// directory in the file system after running. Now the tests clean this up.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/components/executor_test.py [hunk 1]
```diff
@@ -149,10 +153,15 @@ def test_func(output_parameter_path: OutputPath(str)):
                 output_parameter_path,
                 'gs://some-bucket/some_task/nested/output_parameter')
```
Comment (chensun) [inline sdk/python/kfp/components/executor_test.py:154]: Actually, this expected value is wrong here. `OutputPath` expects a local file path, while `OutputUri` could be a URI like `gs://....`.  Can you fix the expected data to be a local path, something like `/gcs/some-bucket/...`?

File: sdk/python/kfp/components/executor_test.py [hunk 2]
```diff
@@ -77,7 +81,7 @@
     },
     "parameters": {
       "output_parameter_path": {
-        "outputFile": "gs://some-bucket/some_task/nested/output_parameter"
+        "outputFile": "gcs/some-bucket/some_task/nested/output_parameter"
```
Comment (chensun) [inline sdk/python/kfp/components/executor_test.py:84]: `/gcs/...`, if I recall correctly, it needs to be an absolute path.

File: sdk/python/kfp/components/executor_test.py [hunk 3]
```diff
@@ -77,33 +81,32 @@
     },
     "parameters": {
       "output_parameter_path": {
-        "outputFile": "gs://some-bucket/some_task/nested/output_parameter"
+        "outputFile": "%(s)s/gcs/some-bucket/some_task/nested/output_parameter"
```
Comment (chensun) [inline sdk/python/kfp/components/executor_test.py:84]: `s` isn't very readable, should we name it to something else? `test_dir`, `root_dir`, or something else?

**Review by chensun — COMMENTED** [review]
/lgtm

File: sdk/python/kfp/components/executor_test.py (hunk 3 above)
Comment (connor-mccarthy) [reply sdk/python/kfp/components/executor_test.py:84]: Definitely -- thanks for the close reviews, @chensun

**Comment by chensun** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**